src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -730,27 +731,52 @@ namespace AppInstaller::CLI::Workflow
                 // The only time we don't want to output a line is when filtering and no update is available.
                 if (updateAvailable || !m_onlyShowUpgrades)
                 {
-                    Utility::LocIndString availableVersion, sourceName;
+                    Utility::LocIndString availableVersion, sourceName, sourceIdentifier;
+                    Utility::LocIndString packageId = match.Package->GetProperty(PackageProperty::Id);

```

Comment:
I would suggest that rather than storing the `packageId` and `sourceIdentifier` separately, just store the `pair` directly.  That would remove 3 times where the strings will be copied to create a temporary pair.